src/util/taproot.rs

```diff
@@ -569,30 +569,30 @@ impl NodeInfo {
     }
 }
 
-// Internally used structure to store information about taproot leaf node
+/// Structure storing information about taproot leaf script.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub(crate) struct LeafInfo {
-    // The underlying script
-    pub(crate) script: Script,
-    // The leaf version
-    pub(crate) ver: LeafVersion,
-    // The merkle proof(hashing partners) to get this node
-    pub(crate) merkle_branch: TaprootMerkleBranch,
+pub struct LeafInfo {
```

Comment:
I am not sure this is easy to review. I agree that this might be good to have, but I will need to review whether allowing setting member fields from the user can break weird invariants in the builder. 

What if the user creates Merkle proofs for two different leaves that are not a part of the same tree. In general, even tough making something `pub` is simple change, making it's members public requires new reasoning and re-reviewing all previous code.

Reply:
Yes, better to provide an accesor methods.